sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp

```diff
@@ -160,16 +160,27 @@ joint_matrix_store(Group sg,
 #endif // __SYCL_DEVICE_ONLY__
 }
 
-template <typename Group, typename T1, typename T2, size_t M, size_t K,
-          size_t N, matrix_layout LayoutA, matrix_layout LayoutB,
+template <typename Group, typename T1, typename T2, typename T3, size_t M,
+          size_t K, size_t N, matrix_layout LayoutA, matrix_layout LayoutB,
           matrix_layout LayoutC>
-inline __SYCL_ALWAYS_INLINE joint_matrix<T2, M, N, LayoutC, Group>
+inline __SYCL_ALWAYS_INLINE joint_matrix<T3, M, N, LayoutC, Group>
 joint_matrix_mad(Group sg, joint_matrix<T1, M, K, LayoutA, Group> &mA,
-                 joint_matrix<T1, K, N, LayoutB, Group> &mB,
-                 joint_matrix<T2, M, N, LayoutC, Group> &mC) {
+                 joint_matrix<T2, K, N, LayoutB, Group> &mB,
+                 joint_matrix<T3, M, N, LayoutC, Group> &mC) {
 #ifdef __SYCL_DEVICE_ONLY__
-  joint_matrix<T2, M, N, LayoutC, Group> res(sg);
-  res.spvm = __spirv_JointMatrixMadINTEL(mA.spvm, mB.spvm, mC.spvm);
+  joint_matrix<T3, M, N, LayoutC, Group> res(sg);
+  if constexpr (std::is_same<T1, uint16_t>::value &&
+                std::is_same<T2, uint16_t>::value &&
+                std::is_same<T3, float>::value)
```

Comment:
Could you please elaborate why this statement is separated from others? If I understand correctly, `__spirv_JointMatrixMadINTEL` should stand for signed-signed Mad, but here it's outlined for unsigned-unsinged mad.

Reply:
According to our previous discussion, If A is bfloat16(aka, uint16), it should choose __spirv_JointMatrixMadINTEL. Am i right? @dkhaldi

Reply:
Right, the unsigned part is relevant 8bits,4,2 bits integers.
uint16_t here is used for bf16.

Reply:
On the other hand, I noticed that the matrix code as currently written does not support half type (FP16).
FP16 should be added. It is not supported by AMX but it is supported by the GPU.